src/precompiles/bn128.rs

```diff
@@ -48,8 +48,11 @@ mod consts {
 pub(super) mod addresses {
     use crate::precompiles;
 
+    #[cfg_attr(not(feature = "engine"), allow(dead_code))]
```

Comment:
Like @mfornet mentioned, we really need to clean up this `engine` feature. What value is it adding?

Reply:
Yes, I think that this initially appeared for some good purposes, but not it's quite confusing, as now we have 74 usages of it in our code. And this is even after I minimized this a little bit with having one statement for several includes in the file. Let's address this in future PRs though.

Reply:
#154 removes the engine flag